src/commands/Music/queue.ts

```diff
@@ -13,22 +15,46 @@ export default class extends MusicCommand {
 	}
 
 	public async run(message: KlasaMessage) {
-		const manager = message.guild!.music;
-		if (!manager.queue.length) throw message.language.tget(manager.song ? 'COMMAND_QUEUE_LAST' : 'COMMAND_QUEUE_EMPTY');
-
-		const output: string[] = [];
-		for (let i = 0; i < Math.min(manager.queue.length, 10); i++) {
-			const song = manager.queue[i];
-			output[i] = [
-				`[__\`${String(i + 1).padStart(2, '0')}\`__] ${message.language.tget(
-					'COMMAND_QUEUE_LINE', song.safeTitle, Util.escapeMarkdown((await song.fetchRequester())!.tag)
-				)}`,
-				`   └── <${song.url}> (${song.friendlyDuration})`
-			].join('\n');
+		const { queue, song } = message.guild!.music;
+		if (!queue.length || !song) throw message.language.tget(message.guild!.music.song ? 'COMMAND_QUEUE_LAST' : 'COMMAND_QUEUE_EMPTY');
+
+		// Send the loading message
+		const response = await message.send(new MessageEmbed()
+			.setColor(getColor(message))
+			.setDescription(message.language.tget('SYSTEM_LOADING')));
+
+		// Format the song entries
+		const songFields = await Promise.all(queue.map(async (song, pos) => message.language.tget('COMMAND_QUEUE_LINE', pos, Util.escapeMarkdown(song.title), song.url, song.friendlyDuration, Util.escapeMarkdown((await song.fetchRequester())!.username))));
+
+		// Generate the pages with 10 songs each
+		const pages = this.chunkify(songFields, 5);
+		const queueDisplay = new UserRichDisplay(new MessageEmbed()
```

Comment:
Do we really want Skyra's specific UserRichDisplay here? I'd rather say make it a regular klasa RichDisplay so anyone can navigate the queue.

Reply:
Yes, we want to. To avoid trolls.

Reply:
fair enough, resolving.